include/swift/AST/DiagnosticsSema.def

```diff
@@ -275,10 +275,10 @@ ERROR(no_candidates_match_argument_type,none,
       (StringRef, Type, unsigned))
 
 ERROR(cannot_infer_closure_parameter_type,none,
-      "unable to infer type of a closure parameter %0 in the current context",
+      "unable to infer type of a closure parameter %0 without a type annotation",
```

Comment:
Maybe `unable to infer type for a closure parameter %0; please provide a type annotation`?

Reply:
If we're going to change the first clause, I would write `cannot infer type of closure parameter %0`

Reply:
Yeah, I think that sounds good and aligns with other diagnostics we have.

Reply:
Maybe I am confused here but I thought we settled on `cannot infer type of closure parameter %0; please provide a type annotation`. A type annotation means that the type-checker can just use the type without having to infer anything.

Reply:
I don't think we settled on that. I was specifically suggesting changing the first clause and leaving what Slava added, i.e. `cannot infer type of closure parameter %0 without a type annotation`.

Reply:
The problem I have with that phrasing is that the type annotation suppresses inference, so asking for annotation to help inference don’t make much sense to me to me personally.

Reply:
`without a type annotation` and `please provide a type annotation` both suggest adding a type annotation, but the former is more concise. I'm generally not a big fan of of using semi-colons in diagnostics, because it encourages making the second clause longer which makes it harder to actually get to the actionable bit of information (e.g. the line annotation is often truncated in IDEs so you have to click to expand it)

In any case, we can always file issues for further improvements because these changes are great first issues for folks. This is already a major improvement over the previous "context" phrasing.

Reply:
I think this phrasing makes it sound like the compiler would attempt to infer the type for the parameter although it would just take the type as is which much give the wrong impression to the users. But I am not a native speaker so I might be wrong here.